### ISSUE TITLE
Is there any option to provide password interactively? 

### ISSUE THREAD

**Issue by raspy** [NuGet/Home#8131]
When setting password from command line (with 'nuget sources') it is visible on the command line and stored in history. Is there any possibility for nuget to ask for password interactively with echo disabled?

**Comment by dquist** [comment]
I think the short answer to this would be 'no'.

Boggles my mind that there's no way to interactively provide the password. This is like command-line authentication 101. Encrypting the password is pointless if it gets entered in plain text and logged in the terminal history.

**Comment by raspy** [comment]
But my point was to make NuGet ask for password, not to provide it on the command line. It will not get stored in terminal history this way.
Note that if you call i.e. _curl -uraspy_ it does not contain a password, but curl will ask interactively for password before issuing actual request. That was my point to do the same with nuget.

**Comment by dquist** [comment]
I'm agreeing with you! We need a way to enter the password so that it won't be stored in the terminal history.

**Comment by jvrobert** [comment]
Yeah, this is a pretty big issue. Went to look through history to find a command in clink, and yep, there are my last 4 passwords I've ever used just sitting in plain text, sweet.
